### DISCUSSION TITLE
Best way to add a countdown clock

### DISCUSSION THREAD

**Discussion by sankalp35** [jspsych/jsPsych#1328] [Q&A]
I want to add a visible countdown clock for every block of my experiment (which is a section of the timeline). I downloaded a jQuery plugin from the internet, tried to add code to each of my "stimulus" parameters, but doesn't look like the clock works. The CSS seems to be overwritten, and the countdown never happens. I'm wondering what's the best way to add a clock to the experiment?

EDIT: After putting the code in "on_load" I was able to get the timer to work. Now my main issue is how do I make the clock tick across block, and not start all over again before every node.

**Comment by jodeleeuw** [comment]
Hi @sankalp35 ,

You can access and modify global variables in the `on_load` function, so if the time remaining is defined globally you can probably configure the timer to show the remaining time.

Another option would be to create the timer on the page external of jsPsych's display element. This would involve using the `display_element` argument in `jsPsych.init` to specify a `<div>` on the page that jsPsych should display in. Then you can have a timer in an element that is not part of the jsPsych `<div>` and it will exist on the page independently of the experiment. You can still modify this timer through jsPsych events like `on_load`.
